### PR TITLE
refactor: use api.at since At() queries are deprecated

### PR DESCRIPTION
**Pull request by gregdhill**
Closes https://github.com/interlay/interbtc-api/issues/353

### CODE REVIEW

File: src/parachain/issue.ts [hunk 1]
```diff
@@ -387,17 +387,18 @@ export class DefaultIssueAPI implements IssueAPI {
...
                     new Promise((resolve, reject) => {
-                        this.api.query.issue.issueRequests
-                            .at(head, ensureHashEncoded(this.api, issueId))
+                        api.query.issue.issueRequests(ensureHashEncoded(this.api, issueId))
                             .then((request) => resolve([request, issueId]))
                             .catch(reject);
                     })
             )
         );
+        // TODO: pass head to parseIssueRequest since it queries chain state
```
Comment (bvotteler) [inline src/parachain/issue.ts:401]: Still todo?

File: src/parachain/redeem.ts [hunk 2]
```diff
@@ -398,16 +398,18 @@ export class DefaultRedeemAPI implements RedeemAPI {
...
+        const api = await this.api.at(head);
 
         const [redeemRequests, redeemPeriod, activeBlockNumber] = await Promise.all([
-            this.api.query.redeem.redeemRequests.entriesAt(head),
+            api.query.redeem.redeemRequests.entries(),
             this.getRedeemPeriod(),
             this.systemAPI.getCurrentActiveBlockNumber(head),
         ]);
-        return await Promise.all(
+        // TODO: pass head to parseRedeemRequest since it queries chain state
```
Comment (bvotteler) [inline src/parachain/redeem.ts:408]: Another todo

File: src/parachain/replace.ts [hunk 3]
```diff
@@ -270,8 +271,10 @@ export class DefaultReplaceAPI implements ReplaceAPI {
 
     async map(): Promise<Map<H256, ReplaceRequestExt>> {
         const head = await this.api.rpc.chain.getFinalizedHead();
-        const replaceRequests = await this.api.query.replace.replaceRequests.entriesAt(head);
+        const api = await this.api.at(head);
+        const replaceRequests = await api.query.replace.replaceRequests.entries();
         const replaceRequestMap = new Map<H256, ReplaceRequestExt>();
+        // TODO: pass head to parseReplaceRequest since it queries chain state
```
Comment (bvotteler) [inline src/parachain/replace.ts:277]: One more todo.

**Review by bvotteler — APPROVED** [review]
LGTM. 
wrt. the TODOs, if not to be completed in this PR, please create a follow up issue.

**Comment by gregdhill** [comment]
Created https://github.com/interlay/interbtc-api/issues/599

**Pull request merged**